flocker/acceptance/test_api.py

```diff
@@ -369,6 +369,61 @@ def can_connect():
         d.addCallback(check_result)
         return d
 
+    def test_create_container_with_environment(self):
+        """
+        Create a container including environment variables on a single-node
+        cluster.
+        """
+        data = {
+            u"name": "my_env_container",
+            u"host": None,
+            u"image": "busybox:latest",
+            u"environment": {u"ACCEPTANCE_ENV_LABEL": 'acceptance test ok'}
+        }
+        waiting_for_cluster = wait_for_cluster(test_case=self, node_count=1)
+
+        def create_container(cluster, data):
+            data[u"host"] = cluster.nodes[0].address
+            return cluster.create_container(data)
+
+        d = waiting_for_cluster.addCallback(create_container, data)
+
+        def check_result((cluster, response)):
+            self.assertEqual(response, data)
+
+        def inspect_container():
+            try:
+                remote_data = run_SSH(
+                    22, 'root', data[u"host"],
+                    [
+                        b"docker",
+                        b"inspect",
+                        b"flocker--{0}".format(data[u"name"])
+                    ], None
+                )
+            except Exception:
+                return False
+            container_data = loads(remote_data)
+            env_data = container_data[0]['Config']['Env']
+            result = dict()
+            for var in env_data:
+                try:
+                    label, value = var.split("=")
+                except ValueError:
+                    continue
+                result[label] = value
+            return result
+
+        def verify_environment(actual, expected):
+            for key, value in expected.iteritems():
```

Comment:
Why not just compare the dictionaries here? That way trial can present a nicely formatted, single failure result in case of item mismatches.

As I said above, I'd move the environment parsing here and then finish with a single `assertEqual` of the expected and actual environment dicts.